integration/src/main/java/scripts/testscripts/CloneWorkspace.java

```diff
@@ -135,7 +144,10 @@ protected void doSetup(
 
     GcsBucketUtils.addFileToBucket(sharedSourceBucket, sourceOwnerUser, sourceProjectId);
 
-    // create a private GCS bucket, which the non-creating user can't clone
+    // Create a folder in the workspace, and add controlled GCS bucket resource to it
```

Comment:
- Let's have addResourceToFolder() take Optional<UUID> for folderId. If folderId is set, addResourceToFolder() will use that folder. If folderId is not set, addResourceToFolder() will create a folder.
- Also, let's have addResourceToFolder() return UUID folderId.

```
UUID folderId = addResourceToFolder(/*folderId=*/ Optional.empty(), sourceOwnerUser, getWorkspaceId(), sharedSourceBucket.getResourceId());
```
For the second addResourceToFolder(), pass in folderId.
- assertFolderInWorkspace() just asserts that folder exists. You need to also assert that resource is in folder. For asserting folder exists, it's easier to use hasItem().
```
assertResourceInFolder(UUID workspaceId, UUID folderID, UUID resourceId) 
{
  // Assert folderId exists
  FolderApi folderApi = new FolderApi(ownerApiClient);
  List<UUID> actualFolderIds = folderApi.listFolders(workspaceId).getFolders().stream().map(...);
  assertThat(actualFolderIds).hasItem(folderID);

  // Assert resource has terra-folder-id property with folderId

}
```